builtin/mainmenu/tab_settings.lua

```diff
@@ -389,6 +423,32 @@ local function handle_settings_buttons(this, fields, tabname, tabdata)
 		core.setting_set("anisotropic_filter", "true")
 		ddhandled = true
 	end
+	
+	if fields["dd_resolution"] == "800 x 600 (4:3)" then
+		core.setting_set("screenW", "800")
+		core.setting_set("screenH", "600")
+		ddhandled = true
+	end
+	if fields["dd_resolution"] == "960 x 540 (16:9)" then
```

Comment:
please use else if .

I also suggest to use a hashtable to set the values, for example { "800 x 600 (4:3)" : ["width": 800, "height": 600] } (it's not the correct syntax for Lua i think but that's the idea)